frontend/src/app/modules/imports/services/data.service.ts

```diff
@@ -17,21 +18,30 @@ import {
   ContentMappingValues,
 } from '../models/mapping.model';
 import { ConfigService } from '@geonature/services/config.service';
+import { ActivatedRoute } from '@angular/router';
 
 @Injectable()
 export class DataService {
   private urlApi = null;
+  private destination: string = null;
 
   constructor(
     private _http: HttpClient,
     public config: ConfigService
   ) {
-    this.urlApi = `${this.config.API_ENDPOINT}/import/synthese`;
-    // this.urlApi = `${this.config.API_ENDPOINT}/import/occhab`;
+    this.urlApi = `${this.config.API_ENDPOINT}/import`;
+  }
+
+  setDestination(destination: string) {
+    this.destination = destination;
+  }
+
+  getUrlApi() {
+    return `${this.urlApi}/${this.destination}`;
   }
```

Comment:
**Suggestion:** The `getUrlApi` function is used to construct the URL for API calls. However, it is not handling the case where `this.destination` is null or undefined. This could potentially lead to incorrect URLs being constructed. You should add a check to ensure `this.destination` is not null or undefined before constructing the URL. [possible issue]
```suggestion
  getUrlApi() {
      if (!this.destination) {
          throw new Error('Destination is not set');
      }
      return `${this.urlApi}/${this.destination}`;
  }
```